File: Python/Easy/flick_switch.py
# https://edabit.com/challenge/YRwZvg5Pkgw4pEWC5
# Create a function that always returns True for every item in a given list.
# However, if an element is the word "flick", switch to always returning the opposite boolean value.
'''
Notes
"flick" will always be given in lowercase.
A list may contain multiple flicks.
Switch the boolean value on the same element as the flick itself.
'''

def flick_switch(lst):
    final = []
    truthy = True
    for item in lst:
        if item == "flick":
            truthy = not(truthy)
            final.append(truthy)
        else:
            final.append(truthy)
    return final

# print(flick_switch(['edabit', 'flick', 'eda', 'bit']), [True, False, False, False])
print(flick_switch([False, False, 'flick', 'sheep', 'flick']))

File: javascript/Medium/flickSwitch.js
// https://edabit.com/challenge/ny2JAEmW5j6NgmpC8

function flickSwitch(arr) {
	let x= true
    for (let i = 0; i < arr.length; i++) {
        if(arr[i]==='flick'){
        x=!x;
        } arr[i]=x
    } return arr
}


console.log(flickSwitch(['edabit', 'flick', 'eda', 'bit']), [true, false, false, false])
console.log(flickSwitch(['flick', 11037, 3.14, 53]), [false, false, false, false])
console.log(flickSwitch([false, false, 'flick', 'sheep', 'flick']), [true, true, false, false, true])
console.log(flickSwitch([flickSwitch]), [true])